lib/ansible/modules/remote_management/manageiq/manageiq_dynamic_resource_definition.py

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) Ahmed Bashir <ahmwag@gmail.com> @ahmbas
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+
+module: manageiq_dynamic_resource_definition
+
+short_description: Management of dynamic resource definitions in ManageIQ.
+extends_documentation_fragment: manageiq
+version_added: '2.6'
+author: Ahmed Bashir (@ahmbas)
+description:
+  - The manageiq_dynamic_resource_definition module supports adding, updating and deleting dynamic resource definitions in ManageIQ.
+
+options:
+  state:
+    description:
+      - absent - dynamic resource definitions should not exist, present - dynamic resource definitions should be.
+    choices: ['absent', 'present']
+    default: 'present'
+  name:
+    description:
+      - The dynamic resource definition's name.
+  properties:
+    description:
+      - The dynamic resource definition's properties.
+    suboptions:
+        attributes:
+          description: dictionary of attributes to pass to the API.
+        associations:
+          description: dictionary with service and provider class.
+        methods:
+          description: list of methods to add to dynamic resource definition.
+  manageiq_connection:
+    description: ManageIQ connection details.
+  version_added: '2.6'
```

Comment:
also remove this line.  you already have it at top level on L21.  
this is what causes the  `'AnsibleUnicode' object has no attribute 'pop'` errors.  here under `options:` ansible-doc assumes all entries will have a sub-hash as value...
(ansible-doc is unhelpful when it crashes, but I run `bin/ansible-doc manageiq_dynamic_resource_definition -vvv` to get traceback, and added print statements from there...)